pkg/resources/marketplace/manifestHelper.go

```diff
@@ -49,6 +68,128 @@ func GenerateRegistryConfigMapFromManifest(manifestPackageName string) (map[stri
+			if info.IsDir() && info.Name() != manifestPackageName {
+				folders = append(folders, path)
+			}
+			return nil
+		})
+
+		ReverseSlice(folders)
+
+		var currentVersion string
+		var err error
+
+		// Get current version from package Yaml
+		if packageYaml != "" {
+			currentVersion, err = GetCurrentCSVFromManifest(packageYaml)
+		}
+		if err != nil {
+			logrus.Fatal("Error")
+		}
+
+		// iterate through all folders
+		for _, currentFolder := range folders {
+
+			// add current version crds
+			if strings.Contains(currentFolder, currentVersion) {
+				e = filepath.Walk(currentFolder, func(path string, info os.FileInfo, err error) error {
+					if err == nil && libRegEx.MatchString(info.Name()) && strings.Contains(path, currentVersion) {
+						return ProcessYamlFile(path, &stringList)
+					}
+					return nil
+				})
+
+			} else {
+				// iterate all other versions look for crds that don't exist in the current version
+				// and add them to the stringlist for building the config map
+				files, err := ioutil.ReadDir(currentFolder)
+				if err == nil {
+					for _, f := range files {
+						// iterate through all fils in the folder
+						if libRegEx.MatchString(f.Name()) {
+							var crdConfig CrdConfig
+							GetCrdDetails(&crdConfig, currentFolder, f)
+							found := CheckFoldersForMatch(folders, currentFolder, &crdConfig, libRegEx)
+							if !found {
+								// if match isn't found, add file contents to stringlist
+								ProcessYamlFile(currentFolder+string(os.PathSeparator)+f.Name(), &stringList)
+							}
+
+						}
+					}
+				}
+			}
+
+		}
+	}
+	return stringList.String(), e
+}
+
+// CheckFoldersForMatch searchs other folders for a crd with the same APIVersion, group and kind
+func CheckFoldersForMatch(folders []string, currentFolder string, crdConfig *CrdConfig, libRegEx *regexp.Regexp) bool {
+
+	found := false
+	for _, folder := range folders {
+		if folder == currentFolder {
+			break
+		} else {
+			files, err := ioutil.ReadDir(folder)
+			if err == nil {
+				for _, f := range files {
+					if libRegEx.MatchString(f.Name()) {
+						var needleCrd CrdConfig
+						GetCrdDetails(&needleCrd, folder, f)
+
+						if crdConfig.APIVersion == needleCrd.APIVersion &&
+							crdConfig.Spec.Group == needleCrd.Spec.Group &&
+							crdConfig.Spec.Names.Kind == needleCrd.Spec.Names.Kind {
+							found = true
+						}
+					}
+				}
+			}
+		}
+	}
+	return found
+}
+
+// GetCrdDetails reads the crd file
+func GetCrdDetails(crdConfig *CrdConfig, currentFolder string, f os.FileInfo) {
+	yamlFile, err := ioutil.ReadFile(currentFolder + string(os.PathSeparator) + f.Name())
+
+	err = yaml.Unmarshal(yamlFile, &crdConfig)
+	if err != nil {
+		fmt.Printf("Error parsing YAML file: %s\n", err)
+	}
+}
+
+// ReverseSlice reverses the order of the folders list
+func ReverseSlice(data interface{}) {
+	value := reflect.ValueOf(data)
+	if value.Kind() != reflect.Slice {
+		panic(errors.New("data must be a slice type"))
```

Comment:
Wouldn't panic crash the whole operator?
I think we should return an error rather then panic.

Reply:
Good point, I've updated that with a new version